### PR TITLE
Sync guest system clock if desynchronized from host

### LINKED ISSUE #1378: hyperkit clock falls behind: failed to write or validate certificate: the certificate is not valid yet
**Minikube version**: v0.18.0

**Environment**:
- **OS**: MacOS Sierra (10.12.4 (16E195))
- **VM Driver** : xhyve
- **ISO version** : v0.18.0

**What happened**:
When I wake my laptop after sleep, the clock of the Minikube VM lags behind. This causes problems with the registry credentials plugin, since AWS rejects credentials requests with an invalid timestamp.

**What you expected to happen**:
The clock should sync after computer wakeup.

**How to reproduce it** (as minimally and precisely as possible):
See above.

### PR DESCRIPTION
**Pull request by tstromberg**
Fixes #1378 and #4264

### CODE REVIEW

**Comment by tstromberg** [comment]
WIP because I would like to write a decent test for this.

**Comment by sharifelgamal** [comment]
Test failure is unrelated to this change.

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -145,11 +158,60 @@ func waitForSSHAccess(h *host.Host, e *engine.Options) error {
...
+		glog.Infof("guest clock delta is within tolerance: %s", d)
+		return nil
+	}
+	if err := adjustGuestClock(h, time.Now()); err != nil {
+		return errors.Wrap(err, "adjusting system clock")
+	}
+	return nil
+}
+
+// systemClockDelta returns the approximate difference between the host and guest system clock
```
Comment (medyagh) [inline pkg/minikube/cluster/cluster.go]: wrong func name in the comment "systemClockDelta" should be "guestClockDelta"

**Pull request merged**